Adding Proxy How-to guide (#72)
rancher/elemental-docs: docs/elemental_behind_proxy.md

:::info info
For this documentation, we assume you are using a SUSE family system (like Elemental Teal), so proxy settings have to be written in `/etc/sysconfig/proxy`.
:::

Proxy settings have to be configured in 3 places:
- Machine Registration Endpoint
- SeedImage resource
- Elemental cluster configuration

## Machine Registration Endpoint
The elemental-system-agent needs proxy settings to reach the Rancher Manager.
To achieve that, we need to fill the cloud-init section of the Machine Registration Endpoint:

You can do it either with [UI](https://elemental.docs.rancher.com/quickstart-ui#add-a-machine-registration-endpoint) or [CLI](https://elemental.docs.rancher.com/quickstart-cli#prepare-your-kubernetes-resources).

<Tabs>
<TabItem value="cliRegistration" label="CLI" default>
<CodeBlock language="yaml" title="registration.yaml" showLineNumbers>{RegistrationProxy}</CodeBlock>
</TabItem>
<TabItem value="uiRegistration" label="UI" default>

![Add proxy settings in Machine Registration](images/proxy-settings-machine-registration-ui.png)
</TabItem>
</Tabs>

## Elemental-register

> [Elemental-register](https://elemental.docs.rancher.com/architecture/#elemental-register-client) is the first communication endpoint between the new host and Rancher Manager, this is the first place where proxy settings could be needed.

Reviewer: ```suggestion
[Elemental-register](https://elemental.docs.rancher.com/architecture/#elemental-register-client) is the first communication endpoint between the new host and Rancher Manager, this is the first place where proxy settings need to be set.
```

:::warning warning

> At the time of writing, that's only possible to configure proxy settings for the ISO in CLI, nothing is implemented in the UI yet ( and we do not know if it will be, ask for it!)

Reviewer: ```suggestion
At the time of writing, it's only possible to configure proxy settings for the ISO with the CLI. The proxy settings aren't implemented in the UI.
```

:::

> The process happens when you boot your Elemental ISO for the first time, in order to configure proxy for it, you have to include a cloud-init definition in the ISO.
> To do that, you have to create a [SeedImage](https://elemental.docs.rancher.com/seedimage-reference/) definition.

Reviewer: ```suggestion
The process happens when you boot your Elemental ISO for the first time, in order to configure the proxy settings you have to include a `cloud-init` definition in the ISO.
To do that, you have to create a [SeedImage](https://elemental.docs.rancher.com/seedimage-reference/) definition.
```

<CodeBlock language="yaml" title="seedimage.yaml" showLineNumbers>{SeedimageProxy}</CodeBlock>

> Apply the YAML with kubectl and then, print your SeedImage definition to get the URL to download it:

Reviewer: ```suggestion
Apply the YAML with `kubectl` and then, print your SeedImage definition to get the URL to download it:
```

```bash showLineNumbers

> kubectl get seedimage <seed_image_name> -n <namespace> -o yaml

Reviewer: As the description before the codeblock mentions an apply, can you add it here too?

```

Boot the ISO and you should see your new system appears in [Machine inventory](https://elemental.docs.rancher.com/architecture#machineinventory).

## Create Elemental cluster

> For that step you can also use either UI or CLI.

Reviewer: ```suggestion

For this step, you can use either the UI or CLI.
```
